compiler/wasm/src/compile_new.rs

```diff
@@ -0,0 +1,258 @@
+        #[allow(deprecated)]
+        let is_opcode_supported = acvm::pwg::default_is_opcode_supported(np_language);
+
+        let root_crate_id = *self.context.root_crate_id();
+
+        let compiled_program =
+            compile_main(&mut self.context, root_crate_id, &compile_options, None, true)
+                .map_err(|errs| {
+                    CompileError::with_file_diagnostics(
+                        "Failed to compile program",
+                        errs,
+                        &self.context.file_manager,
+                    )
+                })?
+                .0;
+
+        let optimized_program =
+            nargo::ops::optimize_program(compiled_program, np_language, &is_opcode_supported)
+                .expect("Program optimization failed");
+
+        let compile_output = preprocess_program(optimized_program);
+        Ok(JsCompileResult::new(compile_output))
+    }
+
+    pub fn compile_contract(mut self) -> Result<JsCompileResult, JsCompileError> {
+        let compile_options = CompileOptions::default();
+        // For now we default to plonk width = 3, though we can add it as a parameter
+        let np_language = acvm::Language::PLONKCSat { width: 3 };
+        #[allow(deprecated)]
+        let is_opcode_supported = acvm::pwg::default_is_opcode_supported(np_language);
+
+        let root_crate_id = *self.context.root_crate_id();
+
+        let compiled_contract =
+            compile_contract(&mut self.context, root_crate_id, &compile_options)
+                .map_err(|errs| {
+                    CompileError::with_file_diagnostics(
+                        "Failed to compile contract",
+                        errs,
+                        &self.context.file_manager,
+                    )
+                })?
+                .0;
+
+        let optimized_contract =
+            nargo::ops::optimize_contract(compiled_contract, np_language, &is_opcode_supported)
+                .expect("Contract optimization failed");
+
+        let compile_output = preprocess_contract(optimized_contract);
+        Ok(JsCompileResult::new(compile_output))
+    }
+}
+
+
+#[cfg(test)]
+mod test {
+    use noirc_driver::prepare_crate;
+    use noirc_frontend::{
+        graph::CrateGraph,
+        hir::Context,
+    };
+
+    use crate::compile::{PathToFileSourceMap, file_manager_with_source_map};
+
+    use std::path::Path;
+
+    use super::ContextWrapper;
+
+    fn setup_test_context(source_map: PathToFileSourceMap) -> ContextWrapper {
+        let mut fm = file_manager_with_source_map(source_map);
+        // Add this due to us calling prepare_crate on "/main.nr" below
+        fm.add_file_with_source(Path::new("/main.nr"), "fn foo() {}".to_string());
+
+        let graph = CrateGraph::default();
+        let mut context = Context::new(fm, graph);
+        prepare_crate(&mut context, Path::new("/main.nr"));
+
+        ContextWrapper{context}
+    }
+
+    #[test]
+    fn test_works_with_empty_dependency_graph() {
```

Comment:
These tests are copied from compile.rs and I just redid them so that we can use the new API above